product/scanning.py

```diff
@@ -118,22 +124,41 @@ def upload_image(
 
         if scan_result:
             # check if image version changed
-
             metadata = scan_result.custom_data()
             image_reference = metadata.get('image-reference')
-            image_changed = image_reference == container_image.image_reference()
+            image_changed = image_reference != container_image.image_reference()
 
-            if not image_changed:
-                # image reference did not change - early exit
+            if not image_changed and upload_mode is UploadMode.UPDATE:
+                # image reference did not change and we need not rescan - early exit
```

Comment:
nit: I think this is noch proper english grammar.
how about: ..no rescanning was requested